mixnet/src/utils.rs

```diff
@@ -0,0 +1,20 @@
+use std::time::Duration;
+
+use rand::thread_rng;
+use rand_distr::{Distribution, Exp};
+
+/// Get a random interval from an exponential distribution.
+pub fn poisson_interval_sec(rate_per_min: f64) -> Duration {
+    // create an exponential distribution
+    let exp = Exp::new(rate_per_min).unwrap();
+    // generate a random value from the distribution
+    let interval_min = exp.sample(&mut thread_rng());
+    // convert minutes to seconds
+    Duration::from_secs_f64(interval_min * 60.0)
+}
```

Comment:
I would say we should abstract the randomness source. For two reasons:
* We may need to have tests that we can use deterministically (hence from a custom seed).
* We may have to use a custom random source for other reasons

Reply:
Yes. It's necessary for testing for sure. The second reason is not really required for mixnet v1 spec, but still it would be worth.

Reply:
@danielSanchezQ Could you please check the commit that Al made [129ef80](https://github.com/logos-co/nomos-node/pull/575/commits/129ef801e05ef0e1545ca0806cd187d2f8ff7b25), so we can merge this? On top of his commits, I pushed additional commits to polish this PR.